dev/devicelab/lib/tasks/web_benchmarks.dart

```diff
@@ -22,9 +22,104 @@ const int benchmarkServerPort = 9999;
 const int chromeDebugPort = 10000;
 
 Future<TaskResult> runWebBenchmark({ @required bool useCanvasKit }) async {
+  // Original (non-gallery) benchmarks.
+  final TaskResult originalBenchmarkResult = await runWebBenchmarkIn(
+    useCanvasKit: useCanvasKit,
+    macrobenchmarksDirectory: path.join(flutterDirectory.path, 'dev', 'benchmarks', 'macrobenchmarks'),
+    entryPoint: 'lib/web_benchmarks.dart',
+  );
+
+  if (originalBenchmarkResult.failed) {
+    return originalBenchmarkResult;
+  }
+
+  // Gallery benchmarks.
+  section('Get New Flutter Gallery!');
+
+  final io.Directory temp = io.Directory.systemTemp.createTempSync('new_gallery_web_benchmarks');
+
+  print('Created temporary directory $temp for Gallery.');
+
+  // TODO(pennzht): When gallery PR merges, move to flutter/gallery. https://github.com/flutter/gallery/issues/326
+  await inDirectory<void>(temp, () async {
+    await exec('git', <String>['clone', 'https://github.com/pennzht/newfluttergallery.git']);
+  });
+
+  print('Cloned into $temp.');
+
+  final io.FileSystemEntity galleryDirectory = temp.listSync().single;
+
+  print('Gallery directory is $galleryDirectory.');
+
+  await inDirectory<void>(galleryDirectory, () async {
+    // TODO(pennzht): When gallery PR merges, use `galleryVersion`. https://github.com/flutter/gallery/issues/326
+    await exec('git', <String>['checkout', '1d285dfe37cc3bd68f2965f8c4186932118d8ee8']);
+    print('Git checkout finished.');
+  });
+
+  final TaskResult galleryBenchmarkResult = await runWebBenchmarkIn(
+    useCanvasKit: useCanvasKit,
+    macrobenchmarksDirectory: galleryDirectory.absolute.path,
+    entryPoint: 'lib/benchmarks/runner.dart',
+  );
+
+  rmTree(temp);
+
+  if (galleryBenchmarkResult.failed) {
+    return galleryBenchmarkResult;
+  }
+
+  // Both succeeded. Combine data.
```

Comment:
What do you think of making this a method on the `TaskResult` class? Something like this:

```dart
class TaskResult {
  TaskResult combine(TaskResult other) {
    ...
  }
}
```

The `combine` method could internally decide whether the overall result is a success or not, and it could validate that there's no overlap in data (e.g. that the benchmark score keys are unique).